### PR TITLE
Fix no visual feedback when updating data model field details

### LINKED ISSUE #20749: "Saved" overlay in data model UI in weird location on smaller screen
Make your browser window a little narrow (about half the screen) and update something in the data model. `Saved` is overlapping the nav bar.

Screen Shot 2022-02-25 at 4 49 51 PM

This was on 41.6, haven't tried on newer versions but I don't think this page has changed recently so opening just in case.

### PR DESCRIPTION
**Pull request by WiNloSt**
A follow-up from PR: #20874, issue #19710

#### After the fix
image
image

@flamber I somehow could get the `Order` ID into a weird state just like the second screenshot, but `undo` button does nothing. I can't reproduce that again once I drop the db and start with a clean slate again. Just FYI.

As far as I know, there shouldn't be much problem undoing fields that are not FK or ID.

### CODE REVIEW

**Review by flamber — APPROVED** [review]
So this works, but if I make multiple changes rapidly, then it seems to indicate that I can undo all of them, but it only goes back one revision. Perhaps that was what you hit too?
image

Given that the toast is hiding so quickly, then I wouldn't mind removing the Undo-button if it causes more problems than good.
The "Saved" status is used everywhere else in Admin > Settings, and it didn't have an Undo, so perhaps we should just replace the "Saved" status with the toast overall?

I'm noticing issue 16780/10462 more, but I think it is unrelated to this PR or your changes, just generally feels like master is slightly slower than previously, but the toast message makes it much more obvious now.

Also, it doesn't seem like there's any status/toast, when changing table name/description/visibility
image

**Comment by WiNloSt** [comment]
@flamber 
> So this works, but if I make multiple changes rapidly, then it seems to indicate that I can undo all of them, but it only goes back one revision. Perhaps that was what you hit too?

No, the issue I encountered is that the undo button is showing for 1 change, but it can't undo for some reason I didn't dig more than that.

> Given that the toast is hiding so quickly, then I wouldn't mind removing the Undo-button if it causes more problems than good.

That's what my initial thought too, but I didn't find much reference to the toast without the undo button (maybe there are, but I just couldn't find it) that was why I decide to go with having the undo button. But seems it started to give us more trouble than helping us, I think it can safely be removed.

> The "Saved" status is used everywhere else in Admin > Settings, and it didn't have an Undo, so perhaps we should just replace the "Saved" status with the toast overall?

I'm not aware of that. To be honest, I just found there is a toast message available when calling API, that's why I used that in the first place. I'm not sure what's the standard message component for a notification like this. Maybe @metabase/frontend-developers could help answer that.

> Also, it doesn't seem like there's any status/toast, when changing table name/description/visibility
Good catch, I only change the code to show a toast message when any field is updated, but I didn't do anything with updating table. I'll modify that as well.

To sum up. 
- I think it's safe to remove `undo` button for success notification on this page, unless there's any concern. That also mean I could revert the code to check circular reference as well which was a hack to make undoing works.
- Add toast message when updating table.

**Comment by WiNloSt** [comment]
@flamber What do you think about the table toast message. I structure the success notification as `Updated Table <field>`
e.g.

image

image

While other fields will have the following success notification as `Updated <field>`, this was due to how we're getting the data to update the field. We got to like the whole field object, so it's not possible to pinpoint the exact field name without refactoring the code which could make it more complicated.
e.g. 
image

image

File: frontend/src/metabase/admin/datamodel/containers/FieldApp.jsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ const mapDispatchToProps = {
   fetchDatabaseMetadata: Databases.actions.fetchDatabaseMetadata,
   fetchTableMetadata: Tables.actions.fetchMetadataAndForeignTables,
   fetchFieldValues: Fields.actions.fetchFieldValues,
-  updateField: Fields.actions.update,
+  updateField: Fields.actions.updateField,
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/containers/FieldApp.jsx:72]: Add notification when updating fields on field details page
image

File: frontend/src/metabase/admin/datamodel/components/database/MetadataTable.jsx [hunk 2]
```diff
@@ -53,7 +53,7 @@ export default class MetadataTable extends Component {
 
   updateProperty(name, value) {
     this.setState({ saving: true });
-    this.props.table.update({ [name]: value });
+    this.props.table.updateProperty(name, value);
```
Comment (WiNloSt) [inline frontend/src/metabase/admin/datamodel/components/database/MetadataTable.jsx:56]: Add notification when updating table `display_name` or `description`.
e.g.

image

image

File: frontend/src/metabase/entities/fields.js [hunk 3]
```diff
@@ -91,7 +91,7 @@ const Fields = createEntity({
       return Fields.actions.update(
         { id: field.id },
         field,
-        undo(opts, field.display_name, t`updated`),
+        notify(opts, field.display_name, t`updated`),
```
Comment (WiNloSt) [inline frontend/src/metabase/entities/fields.js:94]: Change from allowing undoing to only notifying that the change has been made, due to undoing actions seems to cause more trouble than helping us.
https://github.com/metabase/metabase/pull/20915#pullrequestreview-904419004

File: frontend/src/metabase/entities/tables.js [hunk 4]
```diff
@@ -73,6 +74,13 @@ const Tables = createEntity({
 
   // ACTION CREATORS
   objectActions: {
+    updateProperty(entityObject, name, value, opts) {
```
Comment (WiNloSt) [inline frontend/src/metabase/entities/tables.js:77]: Add notification when updating table `display_name` or `description`.
e.g.

image

image

File: frontend/src/metabase/lib/entities.js [hunk 5]
```diff
@@ -259,14 +259,12 @@ export function createEntity(def) {
...
         if (notify) {
-          // pick only the attributes that were updated
-          const undoObject = _.pick(
-            originalObject,
-            ...Object.keys(updatedObject || {}),
-          );
-          // https://github.com/metabase/metabase/pull/20874#pullrequestreview-902384264
-          const canUndo = !hasCircularReference(undoObject);
-          if (notify.undo && canUndo) {
+          if (notify.undo) {
```
Comment (WiNloSt) [inline frontend/src/metabase/lib/entities.js:262]: Revert the code to check circular reference from https://github.com/metabase/metabase/pull/20874 as we don't need to undo changes on the data model page anymore https://github.com/metabase/metabase/pull/20915#pullrequestreview-904419004

**Review by daltojohnso — APPROVED** [review]
There's a component `SaveStatus` that is used when updating other fields in the admin panel, I'm remembering. But using notify seems nice as if provides more info about what you are updating. This seems fine to me. We can always revisit.
